### PR TITLE
Problem:( CRO-496) complex and inflexible initial distribution config

### PR DESCRIPTION
**Pull request by linfeng-crypto**
- change enum `AccountType` to `StakedStateDestination`
- remove struct `InitialValidator` and add `ValidatorPubkey`
- redefine the struct `InitConfig`, remove `launch_incentive_from`,`launch_incentive_to`, `long_term_incentive`, add a field `rewards_pool` to store the initial token amount, change `council_nodes` from a vector into a `BTreeMap`, as below:
```
pub struct InitConfig {
    // initial rewards pool of CRO tokens
    pub rewards_pool: Coin,
    // Redeem mapping of ERC20 snapshot: Eth address => (StakedStateDestination,CRO tokens)
    // (doesn't include the rewards pool amount)
    pub distribution: BTreeMap<RedeemAddress, (StakedStateDestination, Coin)>,
    // initial network parameters
    pub network_params: InitNetworkParameters,
    // initial validators
    pub council_nodes: BTreeMap<RedeemAddress, ValidatorPubkey>,
}
```
- simplify the function `validate_config_get_genesis` for `InitConfig`
- Use new `InitConfig` in `dev-util` and `chain-abci`
- fix test cases

### CODE REVIEW

File: chain-abci/tests/abci_app.rs [hunk 1]
```diff
@@ -800,12 +784,15 @@ fn block_commit(app: &mut ChainNodeApp<MockClient>, tx: TxAux, block_height: i64
...
     println!(
-        "uncommitted root hash: {:?}",
-        app.uncommitted_account_root_hash
+        "uncommitted root hash: {}",
+        hex::encode(app.uncommitted_account_root_hash)
     );
     let account_key = to_stake_key(&StakedStateAddress::from(*account_address));
     let state = app.last_state.clone().expect("app state");
-    println!("committed root hash: {:?}", &state.last_account_root_hash);
+    println!(
```
Comment (leejw51crypto) [inline chain-abci/tests/abci_app.rs]: debug!

Comment (tomtau) [reply chain-abci/tests/abci_app.rs]: this is in tests, so it won't be printed out unless the test fails

Comment (tomtau) [reply chain-abci/tests/abci_app.rs]: so `println!` is OK

File: chain-core/src/init/address.rs [hunk 2]
```diff
@@ -252,6 +252,13 @@ impl fmt::Display for RedeemAddress {
     }
 }
 
+#[cfg(feature = "hex")]
+impl fmt::Debug for RedeemAddress {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        write!(f, "(0x{}", hex::encode(self.0))
+    }
+}
+
```
[the remark this answers is not in the document]
Comment (linfeng-crypto) [reply chain-core/src/init/address.rs]: How about remove the `#[cfg(feature = "hex")]`?

[the remark this answers is not in the document]
Comment (tomtau) [reply chain-core/src/init/address.rs]: it won't compile, because the tx-validation enclave doesn't use chain-core with default features (so there's no hex crate)

**Comment by tomtau** [comment]
bors try

File: chain-core/src/init/address.rs (hunk 2 above)
[the remark this answers is not in the document]
Comment (tomtau) [reply chain-core/src/init/address.rs]: ```error[E0277]: `init::address::RedeemAddress` doesn't implement `core::fmt::Debug`
651	  --> /tmp/drone-ZUPW93uRybLyAqBa/drone/src/chain-core/src/state/account.rs:45:17
652	   |
653	45 |     BasicRedeem(RedeemAddress),
654	   |                 ^^^^^^^^^^^^^ `init::address::RedeemAddress` cannot be formatted using `{:?}`
655	
656	   |
657	   = help: the trait `core::fmt::Debug` is not implemented for `init::address::RedeemAddress`
658	   = note: add `#[derive(Debug)]` or manually implement `core::fmt::Debug`
659	   = note: required because of the requirements on the impl of `core::fmt::Debug` for `&init::address::RedeemAddress`
660	   = note: required for the cast to the object type `dyn core::fmt::Debug`
661	
662	error: aborting due to previous error
663	
664	For more information about this error, try `rustc --explain E0277`.
665	error: Could not compile `chain-core`.```

File: chain-core/src/init/config.rs [hunk 3]
```diff
@@ -309,44 +293,29 @@ impl InitConfig {
...
-            } else {
-                accounts.push(StakedState::new_init(
+    fn get_account(&self, genesis_time: Timespec) -> Vec<StakedState> {
+        self.distribution
+            .iter()
+            .map(|(address, (destination, amount))| {
+                let bonded = match destination {
+                    StakedStateDestination::Bonded => true,
+                    StakedStateDestination::UnbondedFromGenesis => false,
+                    StakedStateDestination::UnbondedFromCustomTime(_time) => false,
```
Comment (tomtau) [inline chain-core/src/init/config.rs]: `UnbondedFromCustomTime` is ignored -- so I guess it'll be implemented in a different PR with `new_init` changes?

File: dev-utils/src/commands/init_command.rs [hunk 4]
```diff
@@ -199,21 +197,6 @@ impl InitCommand {
             "these incentive wallets are special, cannot be withdrawn (anykey)",
```
Comment (tomtau) [inline dev-utils/src/commands/init_command.rs]: this should be also removed -- did you try running the dev-utils commands?

File: chain-core/src/init/config.rs (hunk 3 above)
Comment (linfeng-crypto) [reply chain-core/src/init/config.rs]: yes, I think so.

File: dev-utils/src/commands/init_command.rs (hunk 4 above)
Comment (linfeng-crypto) [reply dev-utils/src/commands/init_command.rs]: no, I'll fix it.

**Comment by tomtau** [comment]
bors try

**Comment by tomtau** [comment]
@linfeng-crypto master changed, so there's a merge conflict with `chain-abci/tests/abci_app.rs`

**Comment by tomtau** [comment]
bors try

**Comment by leejw51crypto** [comment]
i checked enclave hw, lgtm

**Comment by tomtau** [comment]
@linfeng-crypto can you squash the commits (rebase on top of the latest master)?

**Comment by tomtau** [comment]
@linfeng-crypto a conflict in `chain-abci/tests/abci_app.rs` -- just keep the `rewards_pool` in the tests, I guess
